### PR TITLE
Anti-Xray 1.0 new PR

### PR DESCRIPTION
**Pull request by stonar96**
#349

### CODE REVIEW

**Comment by stonar96** [comment]
As I have already mentioned in #349 I will create a new PR for Anti-Xray 2.0 soon. **This is still the old implementation** with the locks just in case that we come back to it again.

**Comment by mibby** [comment]
Any update on Anti-Xray 2.0? :o

**Comment by ScuroK** [comment]
Any news?

File: Spigot-Server-Patches/0167-Anti-Xray.patch [hunk 1]
```diff
@@ -0,0 +1,1541 @@
...
+--- a/src/main/java/net/minecraft/server/Explosion.java
++++ b/src/main/java/net/minecraft/server/Explosion.java
+@@ -233,6 +233,7 @@ public class Explosion {
+                 blockposition = (BlockPosition) iterator.next();
+                 IBlockData iblockdata = this.world.getType(blockposition);
+                 Block block = iblockdata.getBlock();
++                this.world.blockPacketController.updateNearbyBlocks(this.world, blockposition); // Paper - Anti-Xray
+ 
+                 if (flag) {
+                     double d0 = (double) ((float) blockposition.getX() + this.world.random
```
Comment (aikar) [inline Spigot-Server-Patches/0167-Anti-Xray.patch]: This needs to be an obfhelper defined on the same line as the variable c (to ensure it conflicts on updates)

Comment (aikar) [inline Spigot-Server-Patches/0167-Anti-Xray.patch]: All obfuscated accesses here need obf helpers

**Comment by Fonzee9** [comment]
When is this gonna be in paper? :)

**Comment by aikar** [comment]
@Fonzee9  when the changes we have requested are performed and we are happy with the state of the PR.

author has stated intent to do another pass at it as a "2.0". We've not heard anything since. It's not going to be pulled in its current state.

**Comment by Fonzee9** [comment]
@aikar Will this be as laggy as Orebfuscator? Also can you help with a issue with Paper https://github.com/PaperMC/Paper/issues/535

**Comment by RoboMWM** [comment]
@Fonzee9 It's best to keep discussion-related conversation on the forums or IRC instead of hijacking a PR/issue

**Comment by kashike** [comment]
Any update to this?

**Comment by stonar96** [comment]
I am sorry that I didn't reply for a long time. I was busy at university and with lots of other stuff so that I couldn't work on Anti-Xray anymore. As I have already mentioned in the IRC I have decided to continue working on Anti-Xray 2.0 and I have good news. It begins to take shape and I plan to do the first tests in a few days.

**Update:** Today I did the above announced tests and after some bug fixing everything worked fine so far. The new implementation is designed so that the obfuscation is thread safe even without such locks like in Anti-Xray 1.0. There are still some parts to implement (e.g. the configuration part, the block updates and the implementation of the asynchronous thread) but most of the work is done. I plan to PR in a few days.

**Update:** #858

**Comment by zachbr** [comment]
Closing, discussion should move to 858, linked above.

**Pull request closed without merging**